src/main/java/org/eclipse/microprofile/starter/log/LoggingTask.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) 2019 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package org.eclipse.microprofile.starter.log;
+
+import org.eclipse.microprofile.starter.view.EngineData;
+
+import javax.enterprise.inject.spi.CDI;
+
+public class LoggingTask implements Runnable {
+
+    private EngineData engineData;
+
+    public LoggingTask(EngineData engineData) {
+        this.engineData = engineData;
+    }
+
+    @Override
+    public void run() {
+        DynamoDBLogger logger = CDI.current().select(DynamoDBLogger.class).get();
```

Comment:
Why don't you make this class a bean and directly inject this logger?

Reply:
Hmm, that would require refactoring this approach too:

```
...
import javax.enterprise.concurrent.ManagedExecutorService;
...
@ViewScoped
@Named
public class GeneratorDataBean implements Serializable {
...
    @Resource
    private ManagedExecutorService managedExecutorService;
...
---> managedExecutorService.submit(new LoggingTask(engineData));
...
```

Anyway, this is @rdebusscher's design. I did not know how to get rid of "No active contexts for scope type javax.enterprise.context..." while leaving CDI bean with a new thread in the first place.  Rudy?

Reply:
@Emily-Jiang @Karm  I can see if the @Dependent scope works ok here. I'll test this out later.

Reply:
It needs to be @RequestScoped (Threads can be RequestScoped or ApplicationScoped when using in combination with managedExecutorService.
I have Patch for this available.